streams/src/test/java/org/apache/kafka/streams/state/KeyValueStoreTestDriver.java

```diff
@@ -206,8 +206,8 @@ private KeyValueStoreTestDriver(final StateSerdes<K, V> serdes) {
                                       final Serializer<V1> valueSerializer) {
                 // for byte arrays we need to wrap it for comparison
 
-                final K keyTest = serdes.keyFrom(keySerializer.serialize(topic, key));
-                final V valueTest = serdes.valueFrom(valueSerializer.serialize(topic, value));
+                final K keyTest = serdes.keyFrom(keySerializer.serialize(topic, headers, key));
+                final V valueTest = serdes.valueFrom(valueSerializer.serialize(topic, headers, value));
```

Comment:
`StateSerdes` is use to read/write into local stores and changlog topics. Because changelog topic should only be read/written by Kafka Stream, it seems we don't need a change there to me.

Reply:
@mjsax is right. Just to clarify, state store / changelogs today only do header-agnostic serde, so the scope of this PR is only for sink nodes.